command.go

```diff
@@ -24,18 +24,22 @@ var (
 	_ Cmder = (*StringIntMapCmd)(nil)
 	_ Cmder = (*ZSliceCmd)(nil)
 	_ Cmder = (*ScanCmd)(nil)
+	_ Cmder = (*ClusterSlotCmd)(nil)
 )
 
 type Cmder interface {
 	args() []string
 	parseReply(*bufio.Reader) error
 	setErr(error)
+	reset()
 
 	writeTimeout() *time.Duration
 	readTimeout() *time.Duration
+	firstKey() string
 
 	Err() error
 	String() string
+	SetKeyArgPos(int)
```

Comment:
This is ugly. I suggest to
- remove `SetKeyArgPos`
- rename `firstKey` to `clusterKey` to make it clear that it is used to cluster commands
- later we can consider exporting `clusterKey` if user needs it

Instead of

```
 cmd := NewStatusCmd("CONFIG", "SET", parameter, value)
 cmd.SetKeyArgPos(0)
```

use something like

```
 cmd := NewStatusCmd("CONFIG", "SET", parameter, value)
 cmd._base._clusterKeyPos = 0
```